scheduler/src/cook/caches.clj

```diff
@@ -35,4 +35,7 @@
       (.maximumSize (:synthetic-pod-recency-size (config/kubernetes)))
       ; We blocklist a given job from being autoscaled soon after a prior autoscaling.
       (.expireAfterWrite (:synthetic-pod-recency-seconds (config/kubernetes)) TimeUnit/SECONDS)
-      (.build)))
\ No newline at end of file
+      (.build)))
+(mount/defstate ^Cache pool-name->exists?-cache :start (new-cache config/config))
```

Comment:
I agree that they don't need to be a cache, but it's actually quite convenient for them to be a cache for the unit testing path. Querying all pools on startup adds complexity to unit tests, which often use `restore-fresh-database!` in the middle of a test in order to produce a new Datomic connection object.

Reply:
Fair reason. You should make sure to add these caches to flush-caches! though to avoid unit tests depending on accidental configuration sitting in a cache.

Reply:
Good call, added